functions/src/lobby.ts

```diff
@@ -0,0 +1,56 @@
+import * as functions from "firebase-functions";
+import { lobbyCollection, userCollection } from "./firestore-collections";
+import { Lobby } from "./firestore-types/lobby";
+import { UserData } from "./firestore-types/users";
+
+export const startGame = functions.https.onCall(async (data: { code: string }, context) => {
+  // no auth then you shouldn't be here
+  if (context.auth === undefined) {
+    return { error: "Not Signed In" };
+  }
+  // get lobby doc
+  const lobby = await lobbyCollection.doc(data.code).get();
```

Comment:
we don't know that data is an object where code is a string so we should do some validation here.

here is how me and nathan are doing it for avatar a2ff3094d4203331f0dea0ee3f5012b46b5ee47a